lib/src/json_schema/json_schema.dart

```diff
@@ -294,27 +299,61 @@ class JsonSchema {
 
   void _baseResolvePaths() {
     if (_root == this) {
-      _schemaAssignments.forEach((assignment) => assignment());
+      // Validate refs in localRefs.
+      for (Uri localRef in _localRefs) {
+        _getSchemaFromPath(localRef.toString());
+      }
 
-      // filter out requests that can be resolved locally.
+      // Filter out requests that can be resolved locally.
       final List<RetrievalRequest> requestsToRemove = [];
       for (final retrievalRequest in _retrievalRequests) {
+        // Optimistically assume resolution successful, and switch to false on errors.
+        bool resolvedSuccessfully = true;
         JsonSchema localSchema;
-        // check if the ref is actually a standard schema definition, resolve it locally.
-        if (SchemaVersion.fromString(retrievalRequest.schemaUri.toString()) != null) {
-          final definitionRef = retrievalRequest.schemaUri.toString();
-          localSchema = JsonSchema.createSchema(getJsonSchemaDefinitionByRef(definitionRef));
-          _addSchemaToRefMap(retrievalRequest.schemaUri.toString(), localSchema);
-        } else {
-          // attempt to get the schema from the existing schema cache.
-          try {
-            localSchema = _getSchemaFromPath(retrievalRequest.schemaUri.toString());
-          } catch (e) {
-            // DO NOTHING: if we couldn't resolve the path locally,
-            // it just means we need to make a request after all
+        final Uri schemaUri = retrievalRequest.schemaUri;
+
+        // Attempt to resolve schema if it does not exist within ref map already.
+        if (_refMap[schemaUri.toString()] == null) {
+          Uri baseUri;
+          String baseEmptyFragmentUri;
+          if (schemaUri.scheme.isNotEmpty) {
+            baseUri = schemaUri.removeFragment();
+            baseEmptyFragmentUri = '$baseUri#';
+          }
+
+          if (baseUri == _inheritedUri && schemaUri.hasFragment) {
+            // Check if the ref base is the same as in the inherited base.
+            localSchema = _root;
+          } else if (baseEmptyFragmentUri != null && _refMap[baseEmptyFragmentUri] != null) {
+            // Check if the ref base already exists in the _refMap.
+            localSchema = _refMap[baseEmptyFragmentUri];
+          } else if (baseEmptyFragmentUri != null && SchemaVersion.fromString(baseEmptyFragmentUri) != null) {
+            // Check if the ref base is actually a standard schema definition, resolve it locally.
+            localSchema = JsonSchema.createSchema(getJsonSchemaDefinitionByRef(baseEmptyFragmentUri));
+            _addSchemaToRefMap(baseEmptyFragmentUri, localSchema);
+          } else {
+            // Attempt to get the schema from the existing schema cache.
+            try {
+              localSchema = _getSchemaFromPath(retrievalRequest.schemaUri.toString());
```

Comment:
change this to an assert, since it shouldn't be available at this point if we've gotten this far. If we did find it, our above logic would be incorrect.

Reply:
`_getSchemaFromPath` is expected to throw here, so it feels unnecessary to assert that an error is caught. This `else` block now just sets `resolvedSuccessfully` to false